content/english/articles/showcases/tx-indexer/index.md

```diff
@@ -0,0 +1,140 @@
+---
+title: "Showcase: tx-indexer"
+publishDate: 2025-01-29T21:46:00-01:00
+translationKey: "tx-indexer"
+tags: ["gnoland", "gno", "graphql", "jsonrpc"]
+level: Intermediate
+author: wyhaines
+summary: This highlights some of the unique and compelling features of the tx-indexer.
+---
+
+# Showcase: tx-indexer
+
+## Overview
+
+**tx-indexer** is a tool from the Gno team that indexes Tendermint2 chain data and serves it via multiple protocols, including via a GraphQL endpoint. The tx-indexer will allow you to query or subscribe to your chain’s blocks, transactions, or custom message data in real time.
+
+This showcase will 
+1. Introduce the **tx-indexer** and its main functionality.
+2. Give you a quick run-down on how to setup the **tx-indexer** to start ingesting chain data.
+3. Show you how to run a GraphQL query, to whet your appetite for what the tx-indexer can do.
+4. Illustrate one example of how you can leverage the capabilities of tx-indexer via a monitoring tool like Grafana.
+
+---
+
+## What Does tx-indexer Do?
+
+At a high level:
+
+- **Indexes chain data**: It connects to your Gno chain via a specified `--remote` node URL and listens for new blocks and transactions.  
+- **Stores data locally**: It uses PebbleDB to store indexed data on disk.  
+- **Serves that data**: It provides a JSON-RPC 2.0 service (both HTTP and WebSocket) and a GraphQL endpoint to query or subscribe to new events.  
+
+For example, if you’re curious about how many contracts have been deployed on your chain, or how many transactions a certain address has recently made, the **tx-indexer** can help you gather that data. 
+
+### GraphQL!? That Sounds Interesting!
+
+While **tx-indexer** also provides standard JSON-RPC endpoints, the GraphQL interface is particularly convenient for modern web and analytics workflows. You can craft complex queries that filter by message type, block height, or transaction content and get exactly the data you need.
+
+Furthermore, the GraphQL endpoint provides **subscriptions**, which allow you to receive real-time events. This is ideal for dashboards that update with fresh data every few seconds.
+
+---
+
+## Getting Started
+
+### 1. Build and Run
+
+First, clone the repository and build the binary:
+
+```shell
+git clone https://github.com/gnolang/tx-indexer.git
+cd tx-indexer
+make build
+```
+
+Then, run the **tx-indexer** and point it to your Gno chain:
+
+```shell
+./build/tx-indexer start --remote https://rpc.gno.land --db-path indexer-db
+```
+
+This will index data from `https://rpc.test4.gno.land` and store it in `indexer-db`. The GraphQL endpoint (and JSON-RPC endpoint) will be available by default on port `8546` (configurable via the `--listen-address` flag).
+
+> **Note**  
+> The WebSocket endpoint is always at `ws://<listen-address>/ws`. You’ll need the WS port if you plan on receiving subscription events over WebSocket.
```

Comment:
Just a note that the starting protocol is `wss` (WebSockets over SSL/TLS) if they're hosting it somewhere with a certificate (like with https)

Reply:
For local and unsecure deployments it's always going to be `ws`

Reply:
I will add a clarification.